lib/npm/install.js

```diff
@@ -38,24 +41,57 @@ function install(context, next) {
     return next(Error('Install Timed Out'));
   }
 
+  var installOutput = '';
+  var installError = '';
+
   proc.stderr.on('data', function (chunk) {
     context.emit('data', 'warn', 'npm-install:', chunk.toString());
+    if (context.module.stripAnsi) {
+      chunk = stripAnsi(chunk.toString());
+      chunk = chunk.replace(/\r/g, '\n');
+    }
+    installError += chunk;
   });
 
   proc.stdout.on('data', function (chunk) {
     context.emit('data', 'verbose', 'npm-install:', chunk.toString());
+    if (context.module.stripAnsi) {
+      chunk = stripAnsi(chunk.toString());
+      chunk = chunk.replace(/\r/g, '\n');
+    }
+    installOutput += chunk;
   });
 
   proc.on('error', function() {
     bailed = true;
     clearTimeout(timeout);
+    if (!context.testOutput) {
```

Comment:
is there ever a case where this would exist?

If there is an error in install then `npm.test` will never be run